### PR TITLE
refactor: expose as es module

### PR DESCRIPTION
**Pull request by saschanaz**
A maintenance for easier debugging when doing #512.

* This removes 10.x from CI workflow because of no module support, but since this is not dropping the UMD build, it's not a breaking change.
* Refactors `collect()` for module-compatible URL-based approach

Closes #459

### CODE REVIEW

**Comment by saschanaz** [comment]
I think we can drop `dist/` from git and publish it only to npm, what do you think?

**Review by marcoscaceres — APPROVED** [review]
Nice and neat! ❤️

**Comment by saschanaz** [comment]
It turns out that this is actually a breaking change, since the default Node.js behavior for importing CommonJS module in ESM context is to expose the module object as `default` item. I don't think we want to keep this behavior, so a wontfix and a major version bump.

File: package.json [hunk 1]
```diff
@@ -28,11 +27,13 @@
...
     "acquire": "node test/util/acquire.js",
     "build": "npx webpack",
-    "build-debug": "npx webpack --mode none"
+    "build-debug": "npx webpack --mode none",
+    "prepublishOnly": "npm run build"
   },
   "repository": "git://github.com/w3c/webidl2.js",
   "main": "dist/webidl2.js",
-  "module": "index.js",
+  "exports": "./index.js",
```
Comment (ExE-Boss) [inline package.json:35]: Note that this makes it no longer possible to import anything other than `index.js`:
- <https://nodejs.org/api/packages.html#packages_package_entry_points>
- <https://nodejs.org/api/packages.html#packages_conditional_exports>
- <https://nodejs.org/api/packages.html#packages_exports>

Comment (saschanaz) [reply package.json:35]: Good point, and I think we never considered internal modules as stable, so still a good change I guess. Or were you using them?

Comment (ExE-Boss) [reply package.json:35]: It makes `dist/webidl2.js` unimportable in newer **Node** versions.

---

It should probably also be renamed to `dist/webidl2.cjs`, or add a `dist/package.json` file with:
```json
{ "type": "commonjs" }
```

Comment (saschanaz) [reply package.json:35]: Why should it be allowed to be imported in newer Node when there is `index.js`?

**Pull request merged**